### ISSUE TITLE
mpremote: incorrect recursive copying of files under WINDOWS

### ISSUE THREAD

**Issue by IhorNehrutsa** [micropython/micropython#7415]
I try to copy directory to device ESP32 MicroPython v1.15
python.exe mpremote.py connect port:COM8 cp -r IN88 :
```
>ver
Microsoft Windows [Version 10.0.19042.867]

>python.exe mpremote.py connect port:COM8 cp -r IN88 :
cp IN88\angles.py :IN88/IN88\angles.py
cp IN88\boot.py :IN88/IN88\boot.py
cp IN88\esp32_.py :IN88/IN88\esp32_.py
cp IN88\heli.py :IN88/IN88\heli.py
```
Destination file names are incorrect:
cp **IN88**\angles.py :**IN88/IN88**\angles.py

Note: Device os.sep is '/' and WINDOWS os.sep is '\\'

View in Thonny IDE: IN88 directory is Ok, file names are incorrect:
![image](https://user-images.githubusercontent.com/70886343/122559130-98d0ac80-d047-11eb-871e-50e86892a816.png)

Thank you very much.

**Comment by spacemanspiff2007** [comment]
I have the same problem: It's not possible to copy modules.

**Comment by maruno** [comment]
I get the same behaviour, mpremote also can't delete the files anymore itself, getting on OSError

```
$ mpremote.exe ls
ls :
           0 .\lib\logging/
         139 boot.py
$ mpremote.exe rmdir ".\lib\logging/"
rmdir :.\lib\logging/
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
OSError: 39
```

**Comment by dpgeorge** [comment]
Fixed by 1f84440538a017e463aaad9686831ce9527122b5
